src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.NonAndroid.cs

```diff
@@ -393,12 +401,17 @@ private static bool TryGetLocalTzFile([NotNullWhen(true)] out byte[]? rawData, [
             id = null;
             string? tzVariable = GetTzEnvironmentVariable();
 
-            // If the env var is null, use the localtime file
+            // If the env var is null, on iOS/tvOS, grab the default tz from the device.
+            // On all other platforms, use the localtime file.
             if (tzVariable == null)
             {
+#if TARGET_IOS || TARGET_TVOS
+                return TryLoadTzFile(Interop.Sys.GetDefaultTimeZone()!, ref rawData, ref id);
```

Comment:
`TryLoadTzFile` expects a filepath, but `GetDefaultTimeZone` will just return either the timezone name or "Local". Will those files exist on the devices? Otherwise we will still hit setting the timezone to UTC because we are only setting the timezone to UTC when `TryGetLocalTzFile` returns false.

Perhaps we can for iOS just set `id` to `Interop.Sys.GetDefaultTimeZone()` and return true?

Reply:
Maybe we can use

```
try
{
    return FindSystemTimeZoneById (Interop.Sys.GetDefaultTimeZone()!);
}
catch {}
```

right before Returning Utc in `GetLocalTimeZoneFromTzFile` for iOS/tvOS